golang/session_subscribe.go

```diff
@@ -1,13 +1,20 @@
 package main
 
 import (
-	"fmt"
 	"log"
+	"os"
+	"os/signal"
+	"syscall"
 	"time"
 )
 
 func dataPrinter(dataChan <-chan *EndpointData) {
-	for data := range dataChan {
+	for {
+		data, ok := <-dataChan
```

Comment:
I think the for range will stop if channel is closed. So we don't need this change here